src/modules/Exchange.ts

```diff
@@ -109,6 +116,87 @@ export class Exchange {
     );
   }
 
+  public async proxyDepositWithZeroEx(
+    {
+      humanSellAmount,
+      sellTokenAddress,
+      starkKey,
+      positionId,
+      slippagePercentage = '1.0',
+    }: {
+      humanSellAmount: string,
+      sellTokenAddress: string,
+      starkKey: string,
+      positionId: BigNumberable,
+      slippagePercentage?: string,
+    },
+    options?: SendOptions,
+  ): Promise<TxResult> {
+    validateSlippage(slippagePercentage);
+
+    const sellAmount: string = humanCollateralAmountToUint256(humanSellAmount);
+    const isMainnet: boolean = this.contracts.networkId === 1;
+
+    const zeroExRequest = await getZeroExSwapQuote(
+      {
+        sellAmount,
+        sellTokenAddress,
+        buyTokenAddress: isMainnet
+          ? USDC_ADDRESS_MAINNET
+          : DYDX_USDC_ADDRESS_ROPSTEN,
```

Comment:
similar comment as in `src/clients/zeroEx.ts`, I think we should throw an error if we can't support this function on the current network

Reply:
I don't totally agree - I'm happy to do a network to USDC map BUT the asset HAS to be USDC to may as well make this some version of USDC